cmake/OpenCVDetectCUDA.cmake

```diff
@@ -33,7 +33,12 @@ if(CUDA_FOUND)
     if(WIN32)
       find_cuda_helper_libs(nvcuvenc)
     endif()
-    set(HAVE_NVCUVID 1)
+    if(CUDA_nvcuvid_LIBRARY)
+      set(HAVE_NVCUVID 1)
+    endif()
+    if(CUDA_nvcuvenc_LIBRARY)
+      set(HAVE_NVCUVENC 1)
```

Comment:
I think you need to add `#cmakedefine HAVE_NVCUVENC` line to `opencv/cmake/templates/cvconfig.h.in`.